trieste/space.py

```diff
@@ -171,3 +172,19 @@ def discretize(self, num_samples: int) -> DiscreteSearchSpace:
             this :class:`Box`.
         """
         return DiscreteSearchSpace(points=self.sample(num_samples))
+
+    def __mul__(self, box: Box):
+        """
+        Return the Cartesian product of the two :class:`Box`\ es (concatenating their respective lower and upper bounds).
+        :param box: :class:`Box`.
+        :return: the new combined :class:`Box`, or NotImplemented if respective bounds have different dtypes.
```

Comment:
```suggestion
        :return: the new combined :class:`Box`
        :raise TypeError: If the lhs and rhs :class:`Box` bounds have different types.
```
since the user will never actually see a `NotImplemented` (`NotImplemented` is python's under the hood way of deferring `a * b` to `b * a` if it's defined, which it isn't here)